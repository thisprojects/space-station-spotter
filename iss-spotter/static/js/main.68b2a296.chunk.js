(this["webpackJsonpspace-station-spotter"]=this["webpackJsonpspace-station-spotter"]||[]).push([[0],{19:function(e,t,n){e.exports=n(43)},24:function(e,t,n){},43:function(e,t,n){"use strict";n.r(t);var a=n(0),r=n.n(a),o=n(5),c=n.n(o),i=(n(24),n(3)),s=n.n(i),l=n(9),u=n(10),d=n(11),p=n(13),m=n(12),f=n(6),h=n(14),v=n(15),g=function(e){return fetch("https://us-central1-nathan-downes-express-api.cloudfunctions.net/api/maps",{method:"POST",body:"".concat(e)}).then((function(e){return e.json()})).then((function(e){return{coords:e.results[0].geometry.location,address:e.results[0].formatted_address}}))},b=function(e,t){return fetch("https://cors-anywhere.herokuapp.com/http://api.open-notify.org/iss-pass.json?lat=".concat(e,"&lon=").concat(t,"&alt=20&n=3")).then((function(e){return e.json()})).then((function(e){return e.response||null}))},E=function(){var e=Object(l.a)(s.a.mark((function e(t){var n,a,r,o,c,i,l;return s.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:if(n=t,e.t0=!t,!e.t0){e.next=6;break}return e.next=5,new Promise((function(e,t){var n=setTimeout((function(){return t("GeoLocation Time Out")}),1e4);navigator.geolocation.getCurrentPosition((function(t){clearTimeout(n);var a=t.coords.longitude,r=t.coords.latitude;e("".concat(r," ").concat(a))}))}));case 5:n=e.sent;case 6:return e.next=8,g(n);case 8:return a=e.sent,r=a.coords,o=r.lat,c=r.lng,i=a.address,e.next=12,b(o,c);case 12:if(l=e.sent){e.next=15;break}throw new Error("Bad response from space station API");case 15:return e.abrupt("return",{lat:o,lng:c,spaceStationResults:l,address:i});case 16:case"end":return e.stop()}}),e)})));return function(t){return e.apply(this,arguments)}}(),w=n(7),y=n(2),S=function(e){return e<10&&"0".concat(e)||e},O=function(e){return e.results.map((function(e,t){return r.a.createElement("div",{className:"result-row",key:t},r.a.createElement("p",{key:t+1},t+1,r.a.createElement("span",null,r.a.createElement(w.a,{icon:y.a})),function(e){var t=new Date(1e3*e),n=t.getFullYear(),a=["Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"][t.getMonth()],r=t.getDate(),o=S(t.getHours()),c=S(t.getMinutes()),i=S(t.getSeconds());return" ".concat(r," / ").concat(a," / ").concat(n,"  at  ").concat(o,":").concat(c,":").concat(i)}(e.risetime)))}))},k=function(e){var t=e.reset;return r.a.createElement("div",{className:"choose-another-location"},r.a.createElement("button",{onClick:function(){return t(!0)}},"Choose Another Location"))},j=function(e){var t=e.results,n=e.reset,a=e.address;return t&&r.a.createElement("div",{className:"results"},r.a.createElement("p",null,a),r.a.createElement(O,{results:t}),r.a.createElement(k,{reset:n}))},C=function(e){return e.error&&r.a.createElement("div",{className:"location-error"},r.a.createElement("p",null,"Error - Location Unavailable Or Unable To Acquire Geolocation"))||null},L=function(e){return e.loading&&r.a.createElement("div",{className:"loading"},r.a.createElement(w.a,{icon:y.c,size:"3x",spin:!0}))},N=function(e){var t=e.submitLocation;return r.a.createElement("div",{className:"geocode"},r.a.createElement("button",{onClick:function(){return t()}},"Use Current Location"))},x=function(e){function t(){var e,n;Object(u.a)(this,t);for(var a=arguments.length,o=new Array(a),c=0;c<a;c++)o[c]=arguments[c];return(n=Object(p.a)(this,(e=Object(m.a)(t)).call.apply(e,[this].concat(o)))).locationRef=r.a.createRef(),n}return Object(h.a)(t,e),Object(d.a)(t,[{key:"render",value:function(){var e=this;return r.a.createElement(r.a.Fragment,null,r.a.createElement("div",{className:"location-input-wrapper"},r.a.createElement("input",{ref:this.locationRef,id:"location-input",type:"text",placeholder:"Choose Location"})),r.a.createElement("div",{className:"submit-button"},r.a.createElement("button",{onClick:function(){return e.props.submitLocation(e.locationRef.current.value)}},"Submit")))}}]),t}(a.Component),A=function(e){var t=e.overlayState,n=e.error,a=e.submitLocation,o=e.loading;return t&&r.a.createElement("div",{className:"overlay"},r.a.createElement("h1",null,"Space Station Spotter"),r.a.createElement("span",{className:"icon"},r.a.createElement(w.a,{icon:y.b,size:"3x"})),r.a.createElement(N,{submitLocation:a}),r.a.createElement("h2",null,"Or"),r.a.createElement("span",{className:"controls"},r.a.createElement(x,{submitLocation:a})),r.a.createElement("p",null,"When can you see the space station? Enter a location to find out!"),r.a.createElement(C,{error:n}),r.a.createElement(L,{loading:o}))||null},R=function(e){function t(e){var n;return Object(u.a)(this,t),(n=Object(p.a)(this,Object(m.a)(t).call(this,e))).state={lat:52.489471,lng:-1.898575,error:!1,spaceStationResults:null,overlayState:!0,loadingState:!1},n.setOverlayState=n.setOverlayState.bind(Object(f.a)(n)),n.setLocation=n.setLocation.bind(Object(f.a)(n)),n}return Object(h.a)(t,e),Object(d.a)(t,[{key:"setOverlayState",value:function(e){this.setState({overlayState:e})}},{key:"setLocation",value:function(){var e=Object(l.a)(s.a.mark((function e(t){var n,a,r,o,c,i,l=this;return s.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return this.setState({loadingState:!0,error:!1}),e.next=3,E(t).catch((function(){l.setState({error:!0,loadingState:!1})}));case 3:n=e.sent,r=(a=n||{}).lat,o=a.lng,c=a.spaceStationResults,i=a.address,!this.state.error&&this.setState({lat:r,lng:o,spaceStationResults:c,address:i,loadingState:!1,overlayState:!1});case 6:case"end":return e.stop()}}),e,this)})));return function(t){return e.apply(this,arguments)}}()},{key:"render",value:function(){var e=this.state,t=e.lat,n=e.lng,a=e.spaceStationResults,o=e.error,c=e.overlayState,i=e.loadingState,s=e.address;return r.a.createElement("div",{id:"page-wrapper"},r.a.createElement(A,{overlayState:c,submitLocation:this.setLocation,error:o,loading:i}),r.a.createElement("div",{id:"map",className:"map"},r.a.createElement(v.Map,{initialCenter:{lat:52.489471,lng:-1.898575},center:{lat:t,lng:n},google:this.props.google,zoom:3,className:"map"},r.a.createElement(v.Marker,{position:{lat:t,lng:n},name:"Current location"})),r.a.createElement(j,{reset:this.setOverlayState,results:a,address:s})))}}]),t}(a.Component),W=Object(v.GoogleApiWrapper)({apiKey:"AIzaSyCyNv5BOZZfdKO3VDhQCOA3Ufm8tv8rCF8"})(R),T=Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));function U(e,t){navigator.serviceWorker.register(e).then((function(e){e.onupdatefound=function(){var n=e.installing;null!=n&&(n.onstatechange=function(){"installed"===n.state&&(navigator.serviceWorker.controller?(console.log("New content is available and will be used when all tabs for this page are closed. See https://bit.ly/CRA-PWA."),t&&t.onUpdate&&t.onUpdate(e)):(console.log("Content is cached for offline use."),t&&t.onSuccess&&t.onSuccess(e)))})}})).catch((function(e){console.error("Error during service worker registration:",e)}))}c.a.render(r.a.createElement(W,null),document.getElementById("root")),function(e){if("serviceWorker"in navigator){if(new URL(".",window.location.href).origin!==window.location.origin)return;window.addEventListener("load",(function(){var t="".concat(".","/service-worker.js");T?(!function(e,t){fetch(e,{headers:{"Service-Worker":"script"}}).then((function(n){var a=n.headers.get("content-type");404===n.status||null!=a&&-1===a.indexOf("javascript")?navigator.serviceWorker.ready.then((function(e){e.unregister().then((function(){window.location.reload()}))})):U(e,t)})).catch((function(){console.log("No internet connection found. App is running in offline mode.")}))}(t,e),navigator.serviceWorker.ready.then((function(){console.log("This web app is being served cache-first by a service worker. To learn more, visit https://bit.ly/CRA-PWA")}))):U(t,e)}))}}()}},[[19,1,2]]]);
//# sourceMappingURL=main.68b2a296.chunk.js.map