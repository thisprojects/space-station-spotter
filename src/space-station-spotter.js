import React, { Component } from "react";
import { Map, GoogleApiWrapper, Marker } from "google-maps-react";
import {
  callSpaceStationApi,
  getResults,
  key
} from "./js-functions/js-functions";
import { OverLay, DisplayResults } from "./components/components.js";
import "./space-station.css";

export class SpaceStationSpotter extends Component {
  constructor(props) {
    super(props);
    this.state = {
      lat: 52.489471,
      lng: -1.898575,
      locationError: false,
      spaceStationResults: null,
      overlayState: true
    };
  }
  render() {
    const toggleLoadingAnimation = () => {
      let x = document.querySelector(".loading");
      x.classList.toggle("toggle-hide");
    };

    const setOverlayState = state => {
      this.setState({ overlayState: state });
    };

    const setLocation = async source => {
      toggleLoadingAnimation();
      let locationResult = await getResults(source).catch(e => { 
                                                                this.setState({ locationError: true })
                                                                toggleLoadingAnimation();
                                                              });
      // If a valid location is returned, set lat and lng then set call space station api. If location is invalid, display errors.
      if (locationResult && locationResult.lat && locationResult.lng) {
        this.setState({
          lat: locationResult.lat,
          lng: locationResult.lng,
          spaceStationResults: await callSpaceStationApi(
            locationResult.lat,
            locationResult.lng
          ).catch(e => e)
        });
        setOverlayState(false);
        toggleLoadingAnimation();
      } else {
        this.setState({ locationError: true });
      }
    };

    return (
      <div id="page-wrapper">
        <OverLay
          overlayState={this.state.overlayState}
          submitLocation={setLocation}
          locationError={this.state.locationError}
        />
        <div id="map" className="map">
          <Map
            initialCenter={{
              lat: 52.489471,
              lng: -1.898575
            }}
            center={{
              lat: this.state.lat,
              lng: this.state.lng
            }}
            google={this.props.google}
            zoom={3}
            className={"map"}
          >
            <Marker
              position={{ lat: this.state.lat, lng: this.state.lng }}
              name={"Current location"}
            />
          </Map>
          <DisplayResults
            reset={setOverlayState}
            results={this.state.spaceStationResults}
          />
        </div>
      </div>
    );
  }
}

export default GoogleApiWrapper({
  apiKey: key
})(SpaceStationSpotter);
