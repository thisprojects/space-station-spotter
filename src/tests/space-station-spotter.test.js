import React from 'react';
import { SpaceStationSpotter } from '../space-station-spotter.js';
import renderer from 'react-test-renderer';

test('SpaceStationSpotter Matches Snapshot', () => {
  const component = renderer.create(
    <SpaceStationSpotter />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
