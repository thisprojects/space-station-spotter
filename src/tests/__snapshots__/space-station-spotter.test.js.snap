// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpaceStationSpotter Matches Snapshot 1`] = `
<div
  id="page-wrapper"
>
  <div
    className="overlay"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <h1>
      Space Station Spotter
    </h1>
    <span
      className="icon"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-satellite-dish fa-w-16 fa-3x "
        data-icon="satellite-dish"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M305.44954,462.59c7.39157,7.29792,6.18829,20.09661-3.00038,25.00356-77.713,41.80281-176.72559,29.9105-242.34331-35.7082C-5.49624,386.28227-17.404,287.362,24.41381,209.554c4.89125-9.095,17.68975-10.29834,25.00318-3.00043L166.22872,323.36708l27.39411-27.39452c-.68759-2.60974-1.594-5.00071-1.594-7.81361a32.00407,32.00407,0,1,1,32.00407,32.00455c-2.79723,0-5.20378-.89075-7.79786-1.594l-27.40974,27.41015ZM511.9758,303.06732a16.10336,16.10336,0,0,1-16.002,17.00242H463.86031a15.96956,15.96956,0,0,1-15.89265-15.00213C440.46671,175.5492,336.45348,70.53427,207.03078,63.53328a15.84486,15.84486,0,0,1-15.00191-15.90852V16.02652A16.09389,16.09389,0,0,1,209.031.02425C372.25491,8.61922,503.47472,139.841,511.9758,303.06732Zm-96.01221-.29692a16.21093,16.21093,0,0,1-16.11142,17.29934H367.645a16.06862,16.06862,0,0,1-15.89265-14.70522c-6.90712-77.01094-68.118-138.91037-144.92467-145.22376a15.94,15.94,0,0,1-14.79876-15.89289V112.13393a16.134,16.134,0,0,1,17.29908-16.096C319.45132,104.5391,407.55627,192.64538,415.96359,302.7704Z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
    <div
      className="geocode"
    >
      <button
        onClick={[Function]}
      >
        Use Current Location
      </button>
    </div>
    <h2>
      Or
    </h2>
    <span
      className="controls"
    >
      <div
        className="location-input-wrapper"
      >
        <input
          id="location-input"
          placeholder="Choose Location"
          type="text"
        />
      </div>
      <div
        className="submit-button"
      >
        <button
          onClick={[Function]}
        >
          Submit
        </button>
      </div>
    </span>
    <div
      className="location-error"
    />
    <div
      className="loading toggle-hide"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-spinner fa-w-16 fa-spin fa-3x "
        data-icon="spinner"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div
    className="map"
    id="map"
  >
    <div
      className="map"
      style={
        Object {
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "bottom": 0,
            "display": "inherit",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        Loading map...
      </div>
    </div>
    <div />
  </div>
</div>
`;
